ArcGISRuntimeSDKQt_CppSamples/Maps/GenerateOfflineMap_Overrides/GenerateOfflineMap_Overrides.cpp

```diff
@@ -0,0 +1,472 @@
+// [WriteFile Name=GenerateOfflineMap_Overrides, Category=Maps]
+// [Legal]
+// Copyright 2018 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#include "GenerateOfflineMap_Overrides.h"
+
+#include "AuthenticationManager.h"
+#include "Envelope.h"
+#include "FeatureLayer.h"
+#include "GeometryEngine.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "Portal.h"
+#include "PortalItem.h"
+#include "OfflineMapTask.h"
+#include "Point.h"
+
+using namespace Esri::ArcGISRuntime;
+
+const QString GenerateOfflineMap_Overrides::s_webMapId = QStringLiteral("acc027394bc84c2fb04d1ed317aac674");
+
+QString GenerateOfflineMap_Overrides::webMapId() const
```

Comment:
usually we have the constructor as the first function in the cpp

Reply:
Is there a special place for static functions?

Reply:
Ahh, I missed that this was static. Probably fine to leave then